src/core/thread/router_table.hpp

```diff
@@ -369,6 +390,10 @@ class RouterTable : public InstanceLocator, private NonCopyable
     TimeMilli        mRouterIdSequenceLastUpdated;
     uint8_t          mRouterIdSequence;
     uint8_t          mActiveRouterCount;
+#if OPENTHREAD_CONFIG_REFERENCE_DEVICE_ENABLE
+    uint8_t mMinRouterIdAllowed = 0;
+    uint8_t mMaxRouterIdAllowed = Mle::kMaxRouterId;
```

Comment:
Is it better (more clear) to initialize here?

What's the suggested way of doing it in C++?

Reply:
C++11 allows this model of default member variable initialization (though it is not possible for certain types, e.g., bit-fields) and we can mix and match (some variable use default initialization, some from constructor).
It is also possible to include both then the one from constructor wins:

> If a member has a default member initializer and also appears in the member initialization list in a constructor, the default member initializer is ignored for that constructor. [[link](https://en.cppreference.com/w/cpp/language/data_members#Member_initialization)]

In my opinion, it may be better to do all members initializations from one place (easer to see and read). Since we do need the constructor (typically to init the base class), it make sense to do all inits from the constructor.